client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -703,11 +709,18 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
       }
       shuffleAllocatedWorkers.put(shuffleId, allocatedWorkers)
       registeredShuffle.add(shuffleId)
-      commitManager.registerShuffle(shuffleId, numMappers)
+      commitManager.registerShuffle(
+        shuffleId,
+        numMappers,
+        isSegmentGranularityVisible)
 
       // Fifth, reply the allocated partition location to ShuffleClient.
       logInfo(s"Handle RegisterShuffle Success for $shuffleId.")
       val allPrimaryPartitionLocations = slots.asScala.flatMap(_._2._1.asScala).toArray
+      commitManager.registerShuffle(
```

Comment:
Duplicate registerShuffle with line 712-715

Reply:
@codenohup, could you add test case to verify the duplicated registerShuffle is wrong?

Reply:
> @codenohup, could you add test case to verify the duplicated registerShuffle is wrong?

Sure, I'd be happy to add a test case if needed. However, I’m unclear about what you mean by "duplicated registerShuffle is wrong." I believe that registerShuffle is idempotent and should only be called once, and the previous  duplicated registerShuffle were due to our careless mistake.